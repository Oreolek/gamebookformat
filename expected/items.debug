BEGIN DEBUG OUTPUT
Book title: Items
Number of sections: 11


Turn to 1 to begin.
1 (start) -  Demonstrating how to manage player [COLLECT item]Inventory[/COLLECT]. You start the book carrying a [ADD item]sword[/ADD] and a [ADD item]shield[/ADD]. Turn to 2. 
2 (tjunction) -  You have reached a t-junction. Here you find a [FOUND item]key[/FOUND] and a [FOUND item]stick[/FOUND]. You can go west to 11, or east to 7. 
3 (attempt_break_door_with_sword) -  OK. The door is broken, but so is the [DROP item]sword[/DROP]. Turn to 6. 
4 (treasure) -  You found [FOUND item]something valuable[/FOUND], but there is no way forward, so you head back to 2. 
5 (cut_rope) -  OK. That was fun. Turn to 9. 
6 (inside) -  There is a rope here that can be cut using a [HAS item]sword[/HAS]. If you have one and want to do that, see 5. Otherwise turn to 9. 
7 (curse) -  There is a [FOUND item]cursed bracelet[/FOUND] here. You can go on to 10 or go back to 2. You can also drop the [HAS item]stick[/HAS] for no particular reason if you have it, see 8. 
8 (drop_stick) -  OK [DROP item]stick[/DROP] dropped. Turn back to 2 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). 
9 (won) -  Congratulations, you won.
10 (portal) -  A magic portal ahead will only allow you to pass if you did not pick up the [HAS NOT item]cursed bracelet[/HAS NOT], leading you to 4. If you have the [HAS item]cursed bracelet[/HAS] you have to go back to 2 instead. Actually feel free to head back to 2 either way. 
11 (door) -  There is a locked door here. If you have a [HAS item]key[/HAS] you can use that to open the door, see 6. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the [HAS item]sword[/HAS], if you have it, see 3. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see 2. 
END DEBUG OUTPUT
