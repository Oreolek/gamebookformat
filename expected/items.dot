digraph gamebook {

 1->12

1->2
2->7

2->12

2->5
4->3
5->2

5->4

5->3
6->12
7->6

7->12
8->12
9->12

9->7

9->8
10->5
11->10

11->5

11->12
12->9

12->11
}
