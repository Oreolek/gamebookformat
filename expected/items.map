{"drop_stick": 8, "curse": 9, "door": 11, "start": 1, "autotest": 2, "inside": 5, "treasure": 6, "cut_rope": 4, "tjunction": 12, "won": 3, "portal": 7, "attempt_break_door_with_sword": 10}