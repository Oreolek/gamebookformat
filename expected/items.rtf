{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\paperw11900\paperh16840\margl1440\margr1440\vieww14140\viewh14860\viewkind0
\pard\tx566\tx1133\tx1700\tx2267\tx2834\tx3401\tx3968\tx4535\tx5102\tx5669\tx6236\tx6803\ql\qnatural\pardirnatural

\f0\b\fs24 \cf0

\b \qc Items
\b0\

\b Turn to 1 to begin.
\b0\
\
\b \qc 1
\b0\
\ql  Demonstrating how to manage player Inventory. You start the book carrying a sword and a shield. Turn to \b 6
\b0
.  \
\

\b \qc 2
\b0\
\ql  OK. That was fun. Turn to \b 8
\b0
.  \
\

\b \qc 3
\b0\
\ql  There is a rope here that can be cut using a sword. If you have one and want to do that, see \b 2
\b0
. Otherwise turn to \b 8
\b0
.  \
\

\b \qc 4
\b0\
\ql  There is a locked door here. If you have a key you can use that to open the door, see \b 3
\b0
. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the sword, if you have it, see \b 5
\b0
. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see \b 6
\b0
.  \
\

\b \qc 5
\b0\
\ql  OK. The door is broken, but so is the sword. Turn to \b 3
\b0
.  \
\

\b \qc 6
\b0\
\ql  You have reached a t-junction. Here you find a key and a stick. You can go west to \b 4
\b0
, or east to \b 10
\b0
.  \
\

\b \qc 7
\b0\
\ql  A magic portal ahead will only allow you to pass if you did not pick up the cursed bracelet, leading you to \b 11
\b0
. If you have the cursed bracelet you have to go back to \b 6
\b0
 instead. Actually feel free to head back to \b 6
\b0
 either way.  \
\

\b \qc 8
\b0\
\ql  Congratulations, you won. \
\

\b \qc 9
\b0\
\ql  OK stick dropped. Turn back to \b 6
\b0
 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there).  \
\

\b \qc 10
\b0\
\ql  There is a cursed bracelet here. You can go on to \b 7
\b0
 or go back to \b 6
\b0
. You can also drop the stick for no particular reason if you have it, see \b 9
\b0
.  \
\

\b \qc 11
\b0\
\ql  You found something valuable, but there is no way forward, so you head back to \b 6
\b0
.  \
\

}
