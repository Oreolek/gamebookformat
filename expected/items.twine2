<tw-storydata name="Items" startnode="1" creator="Twine" creator-version="2.0" format="Harlowe" options="">
<style role="stylesheet" id="twine-user-stylesheet" type="text/twine-css"></style><script role="script" id="twine-user-script" type="text/twine-javascript"></script>
<tw-passagedata pid="1" name="1" tags="" position="10,10"> Demonstrating how to manage player <b>Inventory</b>. You start the book carrying a <b>sword</b> and a <b>shield</b>. Turn to [[12]]
 or go to the autotest at [[2]]
. </tw-passagedata>
<tw-passagedata pid="2" name="2" tags="" position="20,20"> If you have the <b>stick</b> then you must turn immediately to <b>stick</b>[[12]]
, and everything else in this section is ignored. You find a <b>book</b> here. If you have the <b>sword</b> you must turn immediately to <b>sword</b>[[5]]
, again ignoring everything that follows. You drop the <b>something valuable</b> here if you have it, then go on to [[7]]
.</tw-passagedata>
<tw-passagedata pid="3" name="3" tags="" position="30,30"> Congratulations, you won.</tw-passagedata>
<tw-passagedata pid="4" name="4" tags="" position="40,40"> OK. That was fun. Turn to [[3]]
. </tw-passagedata>
<tw-passagedata pid="5" name="5" tags="" position="50,50"> There is a rope here that can be cut using a <b>sword</b>. If you have one and want to do that, see [[4]]
. You can also run to the autotest at [[2]]
. Otherwise turn to [[3]]
. </tw-passagedata>
<tw-passagedata pid="6" name="6" tags="" position="60,60"> You found <b>something valuable</b>, but there is no way forward, so you head back to [[12]]
. </tw-passagedata>
<tw-passagedata pid="7" name="7" tags="" position="70,70"> A magic portal ahead will only allow you to pass if you did not pick up the <b>cursed bracelet</b>, leading you to [[6]]
. If you have the <b>cursed bracelet</b> you have to go back to [[12]]
 instead. Actually feel free to head back to [[12]]
 either way. </tw-passagedata>
<tw-passagedata pid="8" name="8" tags="" position="80,80"> OK <b>stick</b> dropped. Turn back to [[12]]
 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). </tw-passagedata>
<tw-passagedata pid="9" name="9" tags="" position="90,90"> There is a <b>cursed bracelet</b> here. You can go on to [[7]]
 or go back to [[12]]
. You can also drop the <b>stick</b> for no particular reason if you have it, see [[8]]
. </tw-passagedata>
<tw-passagedata pid="10" name="10" tags="" position="100,100"> OK. The door is broken, but so is the <b>sword</b>. Turn to [[5]]
. </tw-passagedata>
<tw-passagedata pid="11" name="11" tags="" position="110,110"> There is a locked door here. If you have a <b>key</b> you can use that to open the door, see [[5]]
. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the <b>sword</b>, if you have it, see [[10]]
. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see [[12]]
. </tw-passagedata>
<tw-passagedata pid="12" name="12" tags="" position="120,120"> You have reached a t-junction. Here you find a <b>key</b> and a <b>stick</b>. You can go west to [[11]]
, or east to [[9]]
. </tw-passagedata>
</tw-storydata>
