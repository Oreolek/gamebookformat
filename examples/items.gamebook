* 1 start
Demonstrating how to manage player inventory. You start
the book carrying a [take]sword[/take] and a [take]shield[/take].
Turn to [[continue]].

* continue
Here you find a [found]key[/found]. Go on to [[door]].

* door
There is a locked door here.
If you have a [carries]key[/carries] you can use that to
open the door, see [[inside]]. Being right before the link
should be enough for the formatter to figure out that the
key is required to be allowed to follow the link.
Else you can try to open with the [carries]sword[/carries],
if you have it, see [[attempt_break_door_with_sword]].
Hopefully the magic is good enough to pair
pre-conditions to links, or more markup must be added later.
You could also try to go back to pick up the key, see [[continue]].

* attempt_break_door_with_sword
OK. The door is broken, but so is the [drop]sword[/drop].
Turn to [[inside]].

* inside
There is a rope here that can be cut using a [carries]sword[/carries].
If you have one and want to do that, see [[cut_rope]].
Otherwise turn to [[won]].

* cut_rope
OK. That was fun. Turn to [[won]].

* won
Congratulations, you won.