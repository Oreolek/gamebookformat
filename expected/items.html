<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta name="viewport"
        content="width=device-width, initial-scale=1, maximum-scale=1">
        <title>Items</title>
    <script>


    var gamebook = {
        'player' : {
            'started' : false,
            'currentSection' : -1,
            'collections' : {},

            'collect' : function(type, name) {
                this.collections[type] = {
                    'name' : name,
                    'contents' : [],
                    'dropped' : {},
                    'add' : function(what) {
                        if (this.contents.indexOf(what) === -1
                            && !(what in this.dropped)) {
                            this.contents.push(what);
                            this.contents.sort();
                        }
                    },
                    'drop' : function(what) {
                        var i = this.contents.indexOf(what);
                        if (i >= 0) {
                            this.contents.splice(i, 1);
                            this.dropped[what] = true;
                        }
                    },
                    'has' : function(what) {
                        return this.contents.indexOf(what) >= 0;
                    }
                };
                gamebook.addCollectionView(type, name);
            },

            'add' : function(type, what) {
                this.collections[type].add(what);
                gamebook.updateCollectionsView();
            },

            'drop' : function(type, what) {
                this.collections[type].drop(what);
                gamebook.updateCollectionsView();
            },

            'has' : function(type, what) {
                return this.collections[type].has(what);
            },

            'getState' : function() {
                return JSON.stringify({
                    'collections' : this.collections,
                    'currentSection' : this.currentSection,
                    'started' : this.started
                });
            },

            'setState' : function(state) {
            }
        },

        'sections' : {},

        'turnToFunctions' : {},

        'addSection' : function(nr, element) {
            var section = {'element' : element, 'nr' : nr};
            this.sections[nr] = section;
        },

        'turnTo' : function(nr) {
            if (!gamebook.player.started) {
                gamebook.start();
            }
            if (!nr in this.sections) {
                throw new Exception("Can not turn to non-existing section "
                                    + nr + ".");
            }
            this.displaySection(nr);
        },

        'start' : function() {
            this.hideIntroSections();
            this.addClassToClass('startlink', 'nodisplay');
            gamebook.player.started = true;
        },

        'displaySection' : function(nr) {
            if (this.player.currentSection > 0) {
                var section = this.sections[this.player.currentSection];
                section.element.style.display = 'none';
            }
            var e = this.sections[nr].element;
            this.runActions(e.getElementsByClassName('sectiontext')[0]);
            e.style.display = 'block';
            this.player.currentSection = nr;
        },

        'hideIntroSections' : function() {
            this.addClassToClass('introsection', 'nodisplay');
            this.removeClassFromClass('displayintrolink', 'nodisplay');
            this.addClassToClass('hideintrolink', 'nodisplay');
        },

        'showIntroSections' : function() {
            this.runActionsInIntroSections();
            this.removeClassFromClass('introsection', 'nodisplay');
            this.addClassToClass('displayintrolink', 'nodisplay');
            this.removeClassFromClass('hideintrolink', 'nodisplay');
            document.body.scrollIntoView();
        },

        'runActionsInIntroSections' : function() {
            Array.prototype.forEach.call(
                document.getElementsByClassName('introsectionbody'),
                gamebook.runActions);
        },

        'addClassToClass' : function(className, addClass) {
            Array.prototype.forEach.call(
                document.getElementsByClassName(className),
                function(e) {
                    e.classList.add(addClass);
                });
        },

        'removeClassFromClass' : function(className, removeClass) {
            Array.prototype.forEach.call(
                document.getElementsByClassName(className),
                function(e) {
                    e.classList.remove(removeClass);
                });
        },

        'runActions' : function(e) {
            var enableNextLink = true;
            var hasXorScope = false;
            var hasAutoScope = false;
            var xorEnableNext = false;
            var autoDisableAllRemainingLinks = (
                gamebook.player.started && e.classList.contains('introsectionbody'));
            Array.prototype.forEach.call(e.childNodes, function(c) {
                if (!c.classList) {
                    return;
                }
                if (c.classList.contains('sectionref')) {
                    if (enableNextLink && !autoDisableAllRemainingLinks) {
                        gamebook.enableLink(c);
                        if (hasAutoScope) {
                            autoDisableAllRemainingLinks = true;
                        }
                    } else {
                        gamebook.disableLink(c);
                    }
                    enableNextLink = !(hasXorScope && !xorEnableNext);
                    hasAutoScope = false;
                    hasXorScope = false;
                } else if (c.classList.contains('collect')) {
                    gamebook.player.collect(c.dataset.type, c.dataset.name);
                } else if (c.classList.contains('add')) {
                    gamebook.player.add(c.dataset.type, c.dataset.what);
                } else if (c.classList.contains('drop')) {
                    gamebook.player.drop(c.dataset.type, c.dataset.what);
                } else if (c.classList.contains('has')) {
                    enableNextLink = gamebook.player.has(c.dataset.type,
                                                         c.dataset.what);
                } else if (c.classList.contains('hasnot')) {
                    enableNextLink = !gamebook.player.has(c.dataset.type,
                                                          c.dataset.what);
                } else if (c.classList.contains('xor')) {
                    hasXorScope = true;
                    xorEnableNext = !enableNextLink;
                } else if (c.classList.contains('auto')) {
                    hasAutoScope = true;
                } else if (c.classList.contains('random')) {
                    c.addEventListener('click',
                                       gamebook.enableRandomLinkAfter);
                    c.classList.add("enabledlink");
                    c.classList.remove("disabledlink");
                    autoDisableAllRemainingLinks = true;
                }
            });
        },

        'enableLink' : function(e) {
            e.addEventListener('click',
                               gamebook.getTurnToFunction(e.dataset.ref));
            e.classList.add("enabledlink");
            e.classList.remove("disabledlink");
        },

        'disableLink' : function(e) {
            e.removeEventListener('click',
                               gamebook.getTurnToFunction(e.dataset.ref));
            e.classList.remove("enabledlink");
            e.classList.add("disabledlink");
        },

        'enableRandomLinkAfter' : function(event) {
            this.classList.remove("enabledlink");
            this.classList.add("disabledlink");
            var links = [];
            var e = this.nextSibling;
            while (e) {
                if (e.classList && e.classList.contains('sectionref')) {
                    links.push(e);
                }
                e = e.nextSibling;
            }
            if (links.length > 0) {
                var selected = links[Math.floor(Math.random()*links.length)]
                gamebook.enableLink(selected);
            } else {
                console.log("Random with nothing to select?");
            }
            event.preventDefault();
        },

        'addCollectionView' : function(type, name) {
            var ce = document.getElementById('collections');
            var template = document.getElementById('collectionTemplate');
            var e = template.cloneNode(true);
            e.className = "collection";
            e.getElementsByClassName('collectionheading')[0].innerHTML = name;
            e.dataset.type = type;
            ce.appendChild(e);
        },

        'updateCollectionsView' : function() {
            var ce = document.getElementById('collections');
            Array.prototype.forEach.call(ce.childNodes, function(c) {
                if (c.className === 'collection') {
                    var type = c.dataset.type;
                    var collection = gamebook.player.collections[type];
                    var cc = c.getElementsByClassName('collectioncontents')[0];
                    cc.innerHTML = collection.contents.join(', ');
                }
            });
        },

        'getTurnToFunction' : function(nr) {
            if (nr in this.turnToFunctions) {
                return this.turnToFunctions[nr];
            } else {
                var f = function () {
                    gamebook.turnTo(nr);
                };
                this.turnToFunctions[nr] = f;
                return f;
            }
        }

    };

// little hack to make easy to test from node.js
if (typeof exports !== 'undefined') {
    exports.gamebook = gamebook;
}


    </script>
  <style>
    .startlink,.sectionref,.displayintrolink,.hideintrolink,.found,.add,.random {
                   font-weight: bold;
                   margin: 0.2em;
                   vertical-align: middle;
                   white-space: nowrap;
    }

    .sectionref.enabledlink {
        padding-left: 1.5em;
        padding-right: 1.5em;
    }

    .startlink,.enabledlink,.displayintrolink,.hideintrolink,.found,.random {
        cursor: pointer;}

    .startlink,.enabledlink,.displayintrolink,.hideintrolink,.found,
    .random {background: #eef;
    }
    .startlink:hover,.enabledlink:hover,.displayintrolink:hover,
       .hideintrolink:hover {background: #a9f;}
    .disabledlink {color: #bbb; cursor: not-allowed; background: inherit;}
    .disabledlink:hover {background: inherit;}
    .sectionnumber {font-weight: bolder;
                    margin-left: 50%;
                    margin-right: 50%;}
    .section {display: none; width: 90%; margin-left: 5%; margin-right: 5%;}
    .sectiontext {margin-top: 0.5em;}
    .gamebook {max-width: 30em; padding: 1em; width: 100%; font-size: 133%;}
    .collections {margin-top: 4em;}
    .collection {background: #ddd; margin-top: 1em;}
    .collectionheading {}
    .collectionheading::after {content: ": ";}
    .collectioncontents {}
    .collect {}
    .add {font-weight: bold;}
    .drop {font-style: italic;}
    .has {font-style: italic;}
    .hasnot {font-style: italic;}
    .collectionTemplate {display: none;}
    .sectionimage {width: 100%; padding: 1em;}
    .nodisplay {display: none;}
  </style>
  </head>
  <body>
<div class="hideintrolink nodisplay"
    onclick="gamebook.hideIntroSections()">(hide instructions)</div>
  <div class="gamebook">

<div class="startlink"
  onclick="gamebook.turnTo(1)">Turn to 1 to begin.</div>
<div class="section" id="section1">
<span class="sectionnumber" id="para1">1</span>
<div class="sectiontext">
 Demonstrating how to manage player <span class="collect" data-type="item"
    data-name="Inventory">Inventory</span>. You start the book carrying a <span class="add" data-type="item"
      data-what="sword">sword</span> and a <span class="add" data-type="item"
      data-what="shield">shield</span>. Turn to <a class="sectionref enabledlink" data-ref="2">2</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(1, document.getElementById('section1'));
}
</script><div class="section" id="section2">
<span class="sectionnumber" id="para2">2</span>
<div class="sectiontext">
 You have reached a t-junction. Here you find a <span class="found enabledlink"
    onclick="gamebook.player.add('item', 'key')"
>key</span>
 and a <span class="found enabledlink"
    onclick="gamebook.player.add('item', 'stick')"
>stick</span>
. You can go west to <a class="sectionref enabledlink" data-ref="11">11</a>, or east to <a class="sectionref enabledlink" data-ref="7">7</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(2, document.getElementById('section2'));
}
</script><div class="section" id="section3">
<span class="sectionnumber" id="para3">3</span>
<div class="sectiontext">
 OK. The door is broken, but so is the <span class="drop" data-type="item"
      data-what="sword">sword</span>. Turn to <a class="sectionref enabledlink" data-ref="6">6</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(3, document.getElementById('section3'));
}
</script><div class="section" id="section4">
<span class="sectionnumber" id="para4">4</span>
<div class="sectiontext">
 You found <span class="found enabledlink"
    onclick="gamebook.player.add('item', 'something valuable')"
>something valuable</span>
, but there is no way forward, so you head back to <a class="sectionref enabledlink" data-ref="2">2</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(4, document.getElementById('section4'));
}
</script><div class="section" id="section5">
<span class="sectionnumber" id="para5">5</span>
<div class="sectiontext">
 OK. That was fun. Turn to <a class="sectionref enabledlink" data-ref="9">9</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(5, document.getElementById('section5'));
}
</script><div class="section" id="section6">
<span class="sectionnumber" id="para6">6</span>
<div class="sectiontext">
 There is a rope here that can be cut using a <span class="has" data-type="item" data-what="sword">sword</span>. If you have one and want to do that, see <a class="sectionref enabledlink" data-ref="5">5</a>. Otherwise turn to <a class="sectionref enabledlink" data-ref="9">9</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(6, document.getElementById('section6'));
}
</script><div class="section" id="section7">
<span class="sectionnumber" id="para7">7</span>
<div class="sectiontext">
 There is a <span class="found enabledlink"
    onclick="gamebook.player.add('item', 'cursed bracelet')"
>cursed bracelet</span>
 here. You can go on to <a class="sectionref enabledlink" data-ref="10">10</a> or go back to <a class="sectionref enabledlink" data-ref="2">2</a>. You can also drop the <span class="has" data-type="item" data-what="stick">stick</span> for no particular reason if you have it, see <a class="sectionref enabledlink" data-ref="8">8</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(7, document.getElementById('section7'));
}
</script><div class="section" id="section8">
<span class="sectionnumber" id="para8">8</span>
<div class="sectiontext">
 OK <span class="drop" data-type="item"
      data-what="stick">stick</span> dropped. Turn back to <a class="sectionref enabledlink" data-ref="2">2</a> to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(8, document.getElementById('section8'));
}
</script><div class="section" id="section9">
<span class="sectionnumber" id="para9">9</span>
<div class="sectiontext">
 Congratulations, you won.
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(9, document.getElementById('section9'));
}
</script><div class="section" id="section10">
<span class="sectionnumber" id="para10">10</span>
<div class="sectiontext">
 A magic portal ahead will only allow you to pass if you did not pick up the <span class="hasnot" data-type="item"
      data-what="cursed bracelet">cursed bracelet</span>, leading you to <a class="sectionref enabledlink" data-ref="4">4</a>. If you have the <span class="has" data-type="item" data-what="cursed bracelet">cursed bracelet</span> you have to go back to <a class="sectionref enabledlink" data-ref="2">2</a> instead. Actually feel free to head back to <a class="sectionref enabledlink" data-ref="2">2</a> either way. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(10, document.getElementById('section10'));
}
</script><div class="section" id="section11">
<span class="sectionnumber" id="para11">11</span>
<div class="sectiontext">
 There is a locked door here. If you have a <span class="has" data-type="item" data-what="key">key</span> you can use that to open the door, see <a class="sectionref enabledlink" data-ref="6">6</a>. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the <span class="has" data-type="item" data-what="sword">sword</span>, if you have it, see <a class="sectionref enabledlink" data-ref="3">3</a>. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see <a class="sectionref enabledlink" data-ref="2">2</a>. 
</div>
</div>
<script>
if (this.gamebook) {
   gamebook.addSection(11, document.getElementById('section11'));
}
</script>     <div id="collections" class="collections">
    </div>
    <div id="collectionTemplate" class="collectionTemplate">
      <span class="collectionheading"></span>
      <span class="collectioncontents"></span>
    </div>
    </div>
    <script>
if (this.gamebook) {
    gamebook.runActionsInIntroSections();
}
    </script>
<div class="displayintrolink nodisplay"
    onclick="gamebook.showIntroSections()">(show instructions)</div>
    </div>
  </body>
</html>
